File: node/src/app/accesscode/model/entities/accessCode.entity.ts
import { Entity, Column, PrimaryGeneratedColumn, OneToOne, JoinColumn, ManyToOne } from 'typeorm';
import { Visitor } from '../../../visitor/model/entities/visitor.entity';
import { Queue } from '../../../queue/model/entities/queue.entity';
import { Transform } from 'class-transformer';

@Entity({ name: 'accesscode' })
export class AccessCode {
  @Transform(({ value }) => parseInt(value))
  @PrimaryGeneratedColumn()
  id: number;

  @Column('int', { name: 'modificationcounter', nullable: true })
  modificationCounter?: number = 1;

  @Column('timestamp', { name: 'creationtime' })
  creationTime?: Date = new Date();

  @Column('timestamp', { name: 'starttime' })
  startTime?: Date = new Date();

  @Column('timestamp', { name: 'endtime' })
  endTime?: Date = new Date();

  @Transform(({ value }) => parseInt(value))
  @Column('int', { name: 'idvisitor', nullable: true })
  visitorId?: number;

  @Transform(({ value }) => parseInt(value))
  @Column('int', { name: 'idqueue', nullable: true })
  queueId?: number;

  @OneToOne(() => Visitor)
  @JoinColumn({ name: 'id' })
  visitor: Visitor;

  @ManyToOne(() => Queue, queue => queue.accessCodes)
  @JoinColumn({ name: 'idqueue' })
  queue: Queue;
}

File: java-quarkus2/src/main/java/com/devonfw/application/domain/models/AccessCodeEntity.java
package com.devonfw.application.domain.models;

import java.util.Date;

import javax.persistence.*;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "AccessCode")
@Getter
@Setter
@ToString
public class AccessCodeEntity extends ApplicationPersistenceEntity {

  @Temporal(TemporalType.TIMESTAMP)
  private Date creationTime;

  @Temporal(TemporalType.TIMESTAMP)
  private Date startTime;

  @Temporal(TemporalType.TIMESTAMP)
  private Date endTime;

  // , fetch = FetchType.EAGER = eager is evil and you should almost never use it (perf issues, google Hibernate N+1)
  @OneToOne(cascade = CascadeType.DETACH)
  @JoinColumn(name = "idVisitor")
  private VisitorEntity visitor;

  @ManyToOne(cascade = CascadeType.DETACH)
  @JoinColumn(name = "idQueue")
  private QueueEntity queue;
}
